src/math/error.h

```diff
@@ -20,32 +22,46 @@ enum ErrorType
     MAAPEError,
     MAPEError,
     PercentError,
+    ASEError,
     RFactorError,
     EuclideanError,
     nErrorCalculationTypes
 };
+// Error information struct
+struct ErrorReport
+{
+    ErrorType errorType;
+    double value;
+    double firstX;
+    double lastX;
+    int nPointsConsidered;
+};
 // Return enum options for ErrorType
 EnumOptions<Error::ErrorType> errorTypes();
+// Return error report formatted to string
+std::string errorReportString(ErrorReport errorReport);
```

Comment:
Only a simple `struct` I know, but you may as well make this a `const &` to ensure it doesn't get modified in the function.
```suggestion
std::string errorReportString(const ErrorReport &errorReport);
```